plan.go

```diff
@@ -266,13 +266,27 @@ type Change struct {
 	// is either an integer pointing to a child of a set/list, or a string
 	// pointing to the child of a map, object, or block.
 	ReplacePaths []interface{} `json:"replace_paths,omitempty"`
+
+	// BeforeIdentity and AfterIdentity are representations of the resource
+	// identity value both before and after the action.
+	BeforeIdentity interface{} `json:"before_identity,omitempty"`
+	AfterIdentity  interface{} `json:"after_identity,omitempty"`
 }
 
 // Importing is a nested object for the resource import metadata.
 type Importing struct {
 	// The original ID of this resource used to target it as part of planned
 	// import operation.
 	ID string `json:"id,omitempty"`
+
+	// Unknown indicates the ID was unknown at the time of planning. This
+	// would have led to the overall change being deferred, as such this should
+	// only be true when processing changes from the deferred changes list.
+	Unknown bool `json:"unknown,omitempty"`
```

Comment:
```suggestion
	UnknownID bool `json:"unknown,omitempty"`
```

I'm not sure if this is too late of a suggestion - I suppose it is late for changing the raw JSON output (`unknown`) but have you considered `UnknownID` to clarify the fact that this indicates unknowness of the ID rather than identity? I was first confused a bit when reading this.

Reply:
Alternatively - if I misinterpreted the comment and it indicates unknownness in general (for both) then I think the comment needs changing.

Reply:
Nice catch. This can indicate unknownness in general, so I've updated the comment